### ISSUE TITLE
v0.25.1 crashes with ts/all-MiniLM-L12-v2 model

### ISSUE THREAD

**Issue by DmitryKan-TomTom** [typesense/typesense#1368]
## Description

v0.25.1 crashes with ts/all-MiniLM-L12-v2 model
The server has been started with empty data directory.

## Steps to reproduce
Declare the following fields in schema
```

    {
      'name'  :  'catName',
      'type'  :  'string'
    },

    {
      'name'  :  'catDescription',
      'type'  :  'string',
      'optional': true
    },
    {
      "name": "embedding",
      "type": "float[]",
      "embed": {
        "from": [
          "catDescription",
          "catName"
        ],
        "model_config": {
          "model_name": "ts/all-MiniLM-L12-v2"
        }
      }
    }
```

## Expected Behavior

Server accepts writes.

## Actual Behavior

Server crashes with the following trace:

```
I20231108 20:11:50.708155 12212312 raft_server.h:60] Peer refresh succeeded!
I20231108 20:11:58.795127 12212316 text_embedder_manager.cpp:16] Validating and initializing local model: ts/all-MiniLM-L12-v2
W20231108 20:11:58.795342 12212316 http_client.cpp:383] Unable to locate system SSL certificates.
E20231108 20:11:59.537987 12212316 stackprinter.h:50] Typesense crashed...
E20231108 20:11:59.538249 12212316 stackprinter.h:57] 1   libsystem_platform.dylib            0x00007ff818f0e37d _sigtramp + 29
E20231108 20:11:59.538306 12212316 stackprinter.h:57] 2   ???                                 0x000000011b3adc08 0x0 + 4751809544
E20231108 20:11:59.538336 12212316 stackprinter.h:57] 3   typesense-server                    0x00000001060f8131 je_free_default + 2241
E20231108 20:11:59.538369 12212316 stackprinter.h:57] 4   typesense-server                    0x00000001060fdc1f je_free + 2623
E20231108 20:11:59.538393 12212316 stackprinter.h:57] 5   typesense-server                    0x0000000106212415 _ZdlPv + 21
E20231108 20:11:59.538415 12212316 stackprinter.h:57] 6   typesense-server                    0x00000001045a9ba5 _ZN8nlohmann10basic_jsonINSt3__13mapENS1_6vectorENS1_12basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEEbxydS7_NS_14adl_serializerENS3_IhNS7_IhEEEEE5parseIRS9_EESD_OT_NS1_8functionIFbiNS_6detail13parse_event_tERSD_EEEb + 229
E20231108 20:11:59.538439 12212316 stackprinter.h:57] 7   typesense-server                    0x0000000104767461 _ZN19TextEmbedderManager23get_public_model_configERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEE + 1841
E20231108 20:11:59.538465 12212316 stackprinter.h:57] 8   typesense-server                    0x00000001047633d4 _ZN19TextEmbedderManager17init_public_modelERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEE + 372
E20231108 20:11:59.538488 12212316 stackprinter.h:57] 9   typesense-server                    0x0000000104761496 _ZN19TextEmbedderManager29validate_and_init_local_modelERKN8nlohmann10basic_jsonINSt3__13mapENS2_6vectorENS2_12basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEEbxydS8_NS0_14adl_serializerENS4_IhNS8_IhEEEEEERm + 134
E20231108 20:11:59.538511 12212316 stackprinter.h:57] 10  typesense-server                    0x0000000104760945 _ZN19TextEmbedderManager23validate_and_init_modelERKN8nlohmann10basic_jsonINSt3__13mapENS2_6vectorENS2_12basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEEbxydS8_NS0_14adl_serializerENS4_IhNS8_IhEEEEEERm + 325
E20231108 20:11:59.538760 12212316 stackprinter.h:57] 11  typesense-server                    0x000000010467c299 _ZN5field29validate_and_init_embed_fieldERKN3tsl9htrie_mapIcS_NS0_2ah8str_hashIcEEtEERN8nlohmann10basic_jsonINSt3__13mapENSA_6vectorENSA_12basic_stringIcNSA_11char_traitsIcEENSA_9allocatorIcEEEEbxydSG_NS8_14adl_serializerENSC_IhNSG_IhEEEEEERKSM_RS_ + 2745
E20231108 20:11:59.538790 12212316 stackprinter.h:57] 12  typesense-server                    0x000000010467b61e _ZN5field21json_fields_to_fieldsEbRN8nlohmann10basic_jsonINSt3__13mapENS2_6vectorENS2_12basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEEbxydS8_NS0_14adl_serializerENS4_IhNS8_IhEEEEEERSA_RNS4_IS_NS8_IS_EEEE + 366
E20231108 20:11:59.538813 12212316 stackprinter.h:57] 13  typesense-server                    0x00000001046308fd _ZN17CollectionManager17create_collectionERN8nlohmann10basic_jsonINSt3__13mapENS2_6vectorENS2_12basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEEbxydS8_NS0_14adl_serializerENS4_IhNS8_IhEEEEEE + 3997
E20231108 20:11:59.539016 12212316 stackprinter.h:57] 14  typesense-server                    0x0000000104647060 _Z22post_create_collectionRKNSt3__110shared_ptrI8http_reqEERKNS0_I8http_resEE + 336
E20231108 20:11:59.539045 12212316 stackprinter.h:57] 15  typesense-server                    0x00000001045ae874 _ZNSt3__120__packaged_task_funcINS_6__bindIZN14BatchedIndexer3runEvE3$_0JEEENS_9allocatorIS4_EEFvvEEclEv + 2196
E20231108 20:11:59.539067 12212316 stackprinter.h:57] 16  typesense-server                    0x00000001045adc98 _ZNSt3__113packaged_taskIFvvEEclEv + 72
E20231108 20:11:59.539091 12212316 stackprinter.h:57] 17  typesense-server                    0x00000001045adb7e _ZZN10ThreadPoolC1EmENKUlvE_clEv + 446
E20231108 20:11:59.539115 12212316 stackprinter.h:57] 18  typesense-server                    0x00000001045ad94d _ZNSt3__1L14__thread_proxyINS_5tupleIJNS_10unique_ptrINS_15__thread_structENS_14default_deleteIS3_EEEEZN10ThreadPoolC1EmEUlvE_EEEEEPvSA_ + 45
E20231108 20:11:59.539238 12212316 stackprinter.h:57] 19  libsystem_pthread.dylib             0x00007ff818ee1202 _pthread_start + 99
E20231108 20:11:59.539263 12212316 stackprinter.h:57] 20  libsystem_pthread.dylib             0x00007ff818edcbab thread_start + 15
E20231108 20:11:59.539284 12212316 stackprinter.h:77] Generating detailed stack trace...
E20231108 20:11:59.539304 12212316 stackprinter.h:79] _sigtramp (in libsystem_platform.dylib) + 29
0x000000011b3adc08
je_free_default (in typesense-server) + 2241
je_free (in typesense-server) + 2623
operator delete(void*) (in typesense-server) + 21
nlohmann::basic_json<std::__1::map, std::__1::vector, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, long long, unsigned long long, double, std::__1::allocator, nlohmann::adl_serializer, std::__1::vector<unsigned char, std::__1::allocator<unsigned char>>> nlohmann::basic_json<std::__1::map, std::__1::vector, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, long long, unsigned long long, double, std::__1::allocator, nlohmann::adl_serializer, std::__1::vector<unsigned char, std::__1::allocator<unsigned char>>>::parse<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>&>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>&, std::__1::function<bool (int, nlohmann::detail::parse_event_t, nlohmann::basic_json<std::__1::map, std::__1::vector, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, long long, unsigned long long, double, std::__1::allocator, nlohmann::adl_serializer, std::__1::vector<unsigned char, std::__1::allocator<unsigned char>>>&)>, bool) (in typesense-server) + 229
TextEmbedderManager::get_public_model_config(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) (in typesense-server) + 1841
TextEmbedderManager::init_public_model(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) (in typesense-server) + 372
TextEmbedderManager::validate_and_init_local_model(nlohmann::basic_json<std::__1::map, std::__1::vector, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, long long, unsigned long long, double, std::__1::allocator, nlohmann::adl_serializer, std::__1::vector<unsigned char, std::__1::allocator<unsigned char>>> const&, unsigned long&) (in typesense-server) + 134
TextEmbedderManager::validate_and_init_model(nlohmann::basic_json<std::__1::map, std::__1::vector, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, long long, unsigned long long, double, std::__1::allocator, nlohmann::adl_serializer, std::__1::vector<unsigned char, std::__1::allocator<unsigned char>>> const&, unsigned long&) (in typesense-server) + 325
field::validate_and_init_embed_field(tsl::htrie_map<char, field, tsl::ah::str_hash<char>, unsigned short> const&, nlohmann::basic_json<std::__1::map, std::__1::vector, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, long long, unsigned long long, double, std::__1::allocator, nlohmann::adl_serializer, std::__1::vector<unsigned char, std::__1::allocator<unsigned char>>>&, nlohmann::basic_json<std::__1::map, std::__1::vector, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, long long, unsigned long long, double, std::__1::allocator, nlohmann::adl_serializer, std::__1::vector<unsigned char, std::__1::allocator<unsigned char>>> const&, field&) (in typesense-server) + 2745
field::json_fields_to_fields(bool, nlohmann::basic_json<std::__1::map, std::__1::vector, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, long long, unsigned long long, double, std::__1::allocator, nlohmann::adl_serializer, std::__1::vector<unsigned char, std::__1::allocator<unsigned char>>>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>&, std::__1::vector<field, std::__1::allocator<field>>&) (in typesense-server) + 366
CollectionManager::create_collection(nlohmann::basic_json<std::__1::map, std::__1::vector, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, bool, long long, unsigned long long, double, std::__1::allocator, nlohmann::adl_serializer, std::__1::vector<unsigned char, std::__1::allocator<unsigned char>>>&) (in typesense-server) + 3997
post_create_collection(std::__1::shared_ptr<http_req> const&, std::__1::shared_ptr<http_res> const&) (in typesense-server) + 336
std::__1::__packaged_task_func<std::__1::__bind<BatchedIndexer::run()::$_0>, std::__1::allocator<std::__1::__bind<BatchedIndexer::run()::$_0>>, void ()>::operator()() (in typesense-server) + 2196
std::__1::packaged_task<void ()>::operator()() (in typesense-server) + 72
ThreadPool::ThreadPool(unsigned long)::'lambda'()::operator()() const (in typesense-server) + 446
void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, ThreadPool::ThreadPool(unsigned long)::'lambda'()>>(void*) (in typesense-server) + 45
_pthread_start (in libsystem_pthread.dylib) + 99
thread_start (in libsystem_pthread.dylib) + 15

```

After this crash, it is impossible to restart the server (keeps crashing with the same trace).

## Metadata

**Typesense Version**:
0.25.1
**OS**:
macOS Sonoma 14.0
Chip: Apple M1 Pro
Memory: 16 GB

**Comment by jasonbosco** [comment]
Could you try replicating this using Docker?

**Comment by DmitryKey** [comment]
Hey @jasonbosco - replying from a different machine.
I tried with Docker and it works fine. This suggests the issue is in running the binary (0.25.1 and 0.24.1 were tried) directly on Mac.

**Comment by jasonbosco** [comment]
@DmitryKey Thank you for confirming. Since it works with Docker, then this is most likely because of Apple Silicon compatibility issues. We had to drop support for Apple M1 and above in 0.25.x and above because it would have a take non-trivial amount of refactoring to resolve the compatibility issues. So we decided to punt the work for later when we find bandwidth, especially since we now publish ARM Docker builds that run well on Apple Silicon machines.
